File: Session18/11. Container With Most Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        
        l=0
        r=len(height)-1
        maxamount=0
        
        while r>l:
            curramount= min(height[r],height[l])*(r-l)
            maxamount=max(maxamount,curramount)
            if height[r]<height[l]:
                r=r-1
            else:
                l=l+1
            
        return maxamount
        

File: Session18/11. Container With Most Water.java
class Solution {
    public int maxArea(int[] height) {
        int n=height.length;
        int ans=0;
        
        int l=0,r=n-1;
        
        while(l<r)
        {
            int newarea=Math.min(height[l],height[r])*(r-l);
            ans=Math.max(ans,newarea);
            if(height[l]>height[r])
            {
                r--;
            }
            else
            {
                l++;
            }
        }
        
        return ans;
    }
}